app/javascript/react/components/Navbar/MobileHeader.tsx

```diff
@@ -87,20 +90,26 @@ export const MobileHeader = ({
 };
 
 export const MobileCalendarHeader = ({ t }: { t: Function }) => {
-  const { previousUserSettings } = useSelector(selectUserSettingsState);
-
+  const { currentUserSettings, previousUserSettings, searchParams } =
+    useMapParams();
   const navigate = useNavigate();
-  const dispatch = useAppDispatch();
+
+  const handleGoBackClick = useCallback(() => {
+    const newSearchParams = new URLSearchParams(searchParams.toString());
+    newSearchParams.set(
+      UrlParamsTypes.previousUserSettings,
+      currentUserSettings
+    );
+    newSearchParams.set(
+      UrlParamsTypes.currentUserSettings,
+      previousUserSettings
+    );
+    navigate(`${urls.reactMap}?${newSearchParams.toString()}`);
+  }, [currentUserSettings, navigate, previousUserSettings, searchParams]);
```

Comment:
This function is vaguely similar to handle go back click from Mobile header.
I think this can be extracted into reusable utility function and parameterize the navigation URL + optional parameters like `sessionId` and` streamId`. WDYT?